src/v/cluster/health_monitor_backend.cc

```diff
@@ -278,25 +275,11 @@ health_monitor_backend::refresh_cluster_health_cache(force_refresh force) {
     auto holder = _gate.hold();
     auto leader_id = _raft0->get_leader_id();
 
-    // if we are a leader, do nothing
-    if (leader_id == _raft0->self().id()) {
-        co_return errc::success;
-    }
-
     // leadership change, abort old refresh request
     if (_refresh_request && leader_id != _refresh_request->leader_id) {
         abort_current_refresh();
     }
```

Comment:
I know this is old code, but I'm not sure I understand it. This guards against leadership changes that happened up to this point, but we call more async functions before returning the response. Since any node can answer `get_cluster_health_request`, a change of the leader after this point can cause an extra request if we hit a stale follower.

Maybe this check should happen closer to sending the rpc request to the leader.

Reply:
it can not happen under the mutex as it would be acquired by the currently processed request